### DISCUSSION TITLE
Attenuation while someone is speaking

### DISCUSSION THREAD

**Discussion by BamButz** [jagrosh/MusicBot#1482] [Ideas]
This idea was already mentioned in the issues #161, but it seems that it got lost over the time.
I would like to reintroduce this as it is something that many users on my server are asking for.

**Comment by MichailiK** [comment]
Hm, 2 things:
1. I believe I tried implementing this a while back, but couldn't find a way to determine when people started/stopped speaking. I can take another look at this though
2. Depending on the ping to the voice region, there's likely gonna be a noticable delay on when JMusicBot's attenuation kicks in & backs off again.

**Reply by MichailiK** [reply]
Well, looks like past me was quite blind, I've just figured out there is a way to receive audio from users 😅

<img width="659" alt="image" src="https://github.com/jagrosh/MusicBot/assets/39029839/710ea89d-e300-44e8-ae82-e7701165aeb1">

**Comment by MichailiK** [comment]
I have an implementation for this at https://github.com/MichailiK/MusicBot/commit/490a077a038663606ea3f239846aadae5e855c46. Unfortunately, there seems to be a bug in JDA 4 which makes the speaking event only fire once, but then never again, no matter how much you start & stop speaking. I believe we'll have to delay this until JMusicBot 0.5.0, where we will switch to JDA 5.
